corehq/apps/export/forms.py

```diff
@@ -919,30 +921,39 @@ def get_filters(self, can_access_all_locations, accessible_location_ids, group_i
         :param location_ids:
         :param date_range: A DatePeriod or DateSpan
         """
-        form_filters = []
-        if can_access_all_locations:
-            form_filters += [_f for _f in [
-                self._get_group_filter(group_ids),
-                self._get_user_type_filter(user_types),
-            ] if _f]
-
-        form_filters += [_f for _f in [
-            self._get_users_filter(user_ids),
-            self._get_locations_filter(location_ids)
-        ] if _f]
-
-        form_filters = flatten_non_iterable_list(form_filters)
-        if form_filters:
-            form_filters = [OR(*form_filters)]
-        else:
-            form_filters = []
+
+        form_filters = list(filter(None,
+                                   [self._create_user_filter(user_types, user_ids, group_ids, location_ids)]))
         date_filter = self._get_datespan_filter(date_range)
         if date_filter:
             form_filters.append(date_filter)
         if not can_access_all_locations:
             form_filters.append(self._scope_filter(accessible_location_ids))
+
         return form_filters
 
+    def _create_user_filter(self, user_types, user_ids, group_ids, location_ids):
+        all_user_filters = None
+        user_type_filters = self._get_user_type_filters(user_types)
+        user_id_filter = self._get_users_filter(user_ids)
+        group_filter = self._get_group_filter(group_ids)
+        location_filter = self._get_locations_filter(location_ids)
+
+        if not location_filter and not group_filter:
+            group_and_location_metafilter = None
+        elif FILTER_ON_GROUPS_AND_LOCATIONS.enabled(self.domain_object.name) and location_ids and group_ids:
+            group_and_location_metafilter = AND(group_filter, location_filter)
+        else:
+            group_and_location_metafilter = OR(*list(filter(None, [group_filter, location_filter])))
+
+        if group_and_location_metafilter or user_id_filter or user_type_filters:
+            all_user_filter_list = [group_and_location_metafilter, user_id_filter]
+            if user_type_filters:
+                all_user_filter_list += user_type_filters
```

Comment:
Why is this one treated separately?  the `filter(None, ...)` will take care of removing it if necessary, so you can just add it up front.

Reply:
Oh, looks like it returns `None` rather than `[]` if there's nothing selected.  Whatevs.